### DISCUSSION TITLE
thermostat over a valve acting as a cover (with position control)

### DISCUSSION THREAD

**Discussion by FermedePommerieux** [jmcollin78/versatile_thermostat#152] [Ideas]
I use the 'power_percent' sensor to regulate the aperture of the cover, enabling me to control the desired power flow to the circuit.

The thermostat simply opens and closes an 'input_logical' switch. I initially experimented with a 'stat' sensor to calculate the time the switch remains closed (for a 3-minute cycle) and provide feedback on its position. However, I've found that the 'power_percent' sensor works quite effectively.

These valves operate at a relatively slow pace, taking nearly 3 minutes to activate. Such systems are designed for use in large houses with multiple sub-valves, pumps, and sectors.

It would be beneficial to have the capability to configure this system within the dialog configuration and permit the use of position-controlled valves.

**Comment by jmcollin78** [comment]
Hello @FermedePommerieux ,

Since the 3.7.0 release (released this week), you can now control a valve directly through a new Thermostat over valve. I guess this new thermostat type is exactly what you need. You can directly through the configuration UI indicate the entity which controls the valve.
This should be a `number` entity or an `input_number` with a value between 0 and 100 (a percentage).

Tell me if this feature is what you need.
